AzureParallelDeploy/AzureParallelDeploy/src/Deployer.ts

```diff
@@ -56,22 +59,39 @@ export class Deployer {
 				` --resource-group ${this.settings.resourceGroup}` +
 				` --name ${targetService}${slotNameParam}` +
 				` --src "${sourceFiles[0]}"`;
-			let deployment = tl.exec("az", azArgs)
-				.then(
-					result => {
-						console.log(`${service}: ${tl.loc("DeployingServiceOk")}`)
-					},
-					error => {
-						result = false;
-						if (this.debug) {
-							Utility.logError(error);
-						}
-						tl.error(`${service}: ${tl.loc("DeployingServiceError")}`);
-					}
-				);
+
+			this.retryCount[service] = 0;
+
+			let deployment = this.ExecuteDeployment(azArgs, service, result);
 			deployments.push(deployment);
 		});
 		await Promise.all(deployments);
 		return result;
 	}
+
+	public ExecuteDeployment(azArgs: string, service: string, result: boolean): any
+	{
+		return tl.exec("az", azArgs)
+		.then(
+			result => {
+				console.log(`${service}: ${tl.loc("DeployingServiceOk")}`)
+			},
+			error => {
+				let retryCount = this.retryCount[service];
+				if (this.retryCount[service] < Deployer.maxRetries)
+				{
+					this.retryCount[service]++;
+					console.log(`${service}: ${tl.loc("DeployServiceRetry", retryCount)}`)
+					this.ExecuteDeployment(azArgs, service, result);
```

Comment:
Moc sa mi nepáči táto rekurzia, ale budem sa spoliehať, že to funguje tak ako má. 😉

Reply:
Je to riadená rekurzia. Max 3 otočky :)